sdks/java/io/contextual-text-io/src/main/java/org/apache/beam/sdk/io/contextualtextio/RecordWithMetadata.java

```diff
@@ -19,21 +19,45 @@
 
 import com.google.auto.value.AutoValue;
 import org.apache.beam.sdk.annotations.Experimental;
-import org.apache.beam.sdk.annotations.Internal;
 import org.apache.beam.sdk.schemas.AutoValueSchema;
 import org.apache.beam.sdk.schemas.annotations.DefaultSchema;
 
-@Internal
+/**
+ * Helper Class based on {@link AutoValueSchema}, it provides Metadata associated with each Record
+ * when reading from file(s) using {@link ContextualTextIO}.
+ *
+ * <h3>Fields:</h3>
+ *
+ * <ul>
+ *   <li>recordOffset: The offset of a record (the byte at which the record begins) in a file. This
+ *       information can be useful if you wish to reconstruct the file. {@link
+ *       RecordWithMetadata#getRecordOffset()}
+ *   <li>recordNum: The record number of the record from its ordinal file. {@link
+ *       RecordWithMetadata#getRecordNum()}
+ *   <li>recordValue: The value / contents of the records {@link
+ *       RecordWithMetadata#getRecordValue()}
+ *   <li>startingOffset: The starting offset of the range to which the record belongs. {@link
```

Comment:
this name may be confusing with recordOffset. 
Consider:
`rangeOffset`
`recordNumInRange`
`The starting offset of the range (split), which contained the record, when the record was read.`

Reply:
Thanks.
`rangeOffset` seems like a good name.